### PR TITLE
Add scheduled event support and misc

### PR DESCRIPTION
**Pull request by Wolfhound905**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition 

## Description

I am making this a drat because it is not 100% ready to be merged, there are things I do not like and want to change. As of now, this has support for mostly everything. There are some checks missing when editing an event and similar stuff like that. I will keep this up to date. Also, there are a few other small changes that are part of this like Timestamp object support in channel get messages.

## Changes

- Scheduled Event Model
- Scheduled Event Requests
- Scheduled Event Functionality
- Timestamp in `channel.get_messages()`

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [ ] I've ensured my code works on `Python 3.10.x`
- [ ] I've tested my code

### CODE REVIEW

File: dis_snek/http_requests/scheduled_events.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        parameters:
+            guild_id: The guild to modify scheduled event from
+            scheduled_event_id: The scheduled event to modify
+            payload: The payload to modify the scheduled event with
+            reason: The reason to be displayed in audit logs
+        returns:
+            Scheduled Event or None
+        """
+        return await self.request(
+            Route("PATCH", f"/guilds/{guild_id}/scheduled-events/{scheduled_event_id}"), data=payload, reason=reason
```
Comment (KAJdev) [inline dis_snek/http_requests/scheduled_events.py:78]: ?

**Comment by LordOfPolls** [comment]
This commit history is a mess. 

If you were just trying to merge the Scheduled Events I would be fine with merging it (tests pending). But the scope of this PR is now scheduled events with various timestamp changes snuck in. 

You should have kept the timestamp pr and made scheduled events their own pr

**Comment by Wolfhound905** [comment]
> This commit history is a mess.
> 
> If you were just trying to merge the Scheduled Events I would be fine with merging it (tests pending). But the scope of this PR is now scheduled events with various timestamp changes snuck in.
> 
> You should have kept the timestamp pr and made scheduled events their own pr

Ya I remember I had that pr open right after I pushed those changes, I just wasn't comfortable reverting. I will remember for next time. Sorry bout that 😅

**Comment by Wolfhound905** [comment]
I really dont like all those commits, time to do better this time

**Pull request closed without merging**